packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -402,9 +402,11 @@ export class Viewer {
 
       await this.waitNextDrawnFrame(15 * 1000);
     } catch (e) {
-      this.errorMessage = 'Unable to establish connection to Vertex.';
-      console.error('Failed to establish WS connection', e);
-      throw new WebsocketConnectionError(this.errorMessage, e);
+      if (this.lastFrame == null) {
+        this.errorMessage = 'Unable to establish connection to Vertex.';
+        console.error('Failed to establish WS connection', e);
+        throw new WebsocketConnectionError(this.errorMessage, e);
+      }
```

Comment:
what is the scenario where `lastFrame` is `null` here? and why do we only want to throw the exception when it is?

Reply:
last frame is null initially, and in between unloading & reloading a scene when a frame gets returned. If the user has a frame.

Reply:
Why is this check necessary?

Reply:
In the event that you spam switching scenes, one of the failed promises will throw an exception here because a promise times out. This only happens when switching between 3+ scenes at once in rapid succession. I think cancelling connection requests or wrapping the connection promise in some disposable would be the long term resolution we'd want to have.